// @mongez/cache/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", { value: true });

var CacheManager = require("./CacheManager.js");
var config = require("./config.js");
var BaseCacheEngine = require("./drivers/BaseCacheEngine.js");
var EncryptedLocalStorageDriver = require("./drivers/EncryptedLocalStorageDriver.js");
var EncryptedSessionStorageDriver = require("./drivers/EncryptedSessionStorageDriver.js");
var PlainLocalStorageDriver = require("./drivers/PlainLocalStorageDriver.js");
var PlainSessionStorageDriver = require("./drivers/PlainSessionStorageDriver.js");
var RunTimeDriver = require("./drivers/RunTimeDriver.js");

exports.CacheManager = CacheManager.CacheManager;
exports["default"] = CacheManager["default"];
exports.getCacheConfig = config.getCacheConfig;
exports.getCacheConfigurations = config.getCacheConfigurations;
exports.setCacheConfigurations = config.setCacheConfigurations;
exports.BaseCacheEngine = BaseCacheEngine;
exports.EncryptedLocalStorageDriver = EncryptedLocalStorageDriver;
exports.EncryptedSessionStorageDriver = EncryptedSessionStorageDriver;
exports.PlainLocalStorageDriver = PlainLocalStorageDriver;
exports.PlainSessionStorageDriver = PlainSessionStorageDriver;
exports.RunTimeDriver = RunTimeDriver;
