// @mongez/encryption/esm/encryption.js
import CryptoJS from 'crypto-js';
import { getEncryptionConfig } from './configurations.js';

/**
 * Return sha1 hashed string
 *
 * @param {string} text
 * @returns {string}
 */
function md5(text) {
    return CryptoJS.MD5(text).toString();
}
/**
 * Return sha1 hashed string
 *
 * @param {string} text
 * @returns {string}
 */
function sha1(text) {
    return CryptoJS.SHA1(text).toString();
}
/**
 * Return sha256 hashed string
 *
 * @param {string} text
 * @returns {string}
 */
function sha256(text) {
    return CryptoJS.SHA256(text).toString();
}
/**
 * Return sha512 hashed string
 *
 * @param {string} text
 * @returns {string}
 */
function sha512(text) {
    return CryptoJS.SHA512(text).toString();
}
/**
 * Get the encrypted text of the given value
 *
 * @param {any} value
 * @param {string} key
 * @param {any} driver
 * @returns {string}
 */
function encrypt(value, key, driver) {
    if (key === void 0) { key = getEncryptionConfig("key"); }
    if (driver === void 0) { driver = getEncryptionConfig("driver"); }
    if (!key) {
        throw new Error("Missing Encryption key, please define it or set it in encryption configurations");
    }
    var data = JSON.stringify({
        data: value,
    });
    return driver.encrypt(data, key).toString();
}
/**
 * Decrypt the given cypher text and return its original value, otherwise null will be returned.
 *
 * @param {string} cypher
 * @param {string} key
 * @param {any} driver
 * @returns {string|null}
 */
function decrypt(cypher, key, driver) {
    if (key === void 0) { key = getEncryptionConfig("key"); }
    if (driver === void 0) { driver = getEncryptionConfig("driver"); }
    if (!key) {
        throw new Error("Missing Encryption key, please define it or set it in encryption configurations");
    }
    try {
        var value = driver.decrypt(cypher, key).toString(CryptoJS.enc.Utf8);
        if (!value)
            return null;
        return JSON.parse(value).data;
    }
    catch (error) {
        console.warn(error);
        return null;
    }
}

export { decrypt, encrypt, md5, sha1, sha256, sha512 };
